dipy/reconst/shm.py

```diff
@@ -230,164 +232,205 @@ def real_sph_harm(m, n, theta, phi):
 
+    real_sh = np.where(m > 0, sh.imag, sh.real)
+    real_sh *= np.where(m == 0, 1., np.sqrt(2))
 
-def real_sym_sh_basis(sh_order, theta, phi):
-    """Samples a real symmetric spherical harmonic basis at point on the sphere
+    return real_sh, m, n
 
-    Samples the basis functions up to order `sh_order` at points on the sphere
-    given by `theta` and `phi`. The basis functions are defined here the same
-    way as in Descoteaux et al. 2007 [1]_ where the real harmonic $Y^m_n$ is
-    defined to be:
 
-        Imag($Y^m_n$) * sqrt(2)     if m > 0
-        $Y^0_n$                     if m = 0
-        Real($Y^|m|_n$) * sqrt(2)   if m < 0
+def real_sym_sh_mrtrix(sh_order, theta, phi):
+    """
+    Compute real symmetric spherical harmonics as in Tournier 2007 [2]_, where
+    the real harmonic $Y^m_n$ is defined to be::
+
+        Real($Y^m_n$)       if m > 0
+        $Y^0_n$             if m = 0
+        Imag($Y^|m|_n$)     if m < 0
 
     This may take scalar or array arguments. The inputs will be broadcasted
     against each other.
 
     Parameters
     -----------
     sh_order : int
-        even int > 0, max spherical harmonic degree
-    theta : float [0, 2*pi]
-        The azimuthal (longitudinal) coordinate.
-    phi : float [0, pi]
+        The maximum degree or the spherical harmonic basis.
+    theta : float [0, pi]
         The polar (colatitudinal) coordinate.
+    phi : float [0, 2*pi]
+        The azimuthal (longitudinal) coordinate.
 
     Returns
     --------
     y_mn : real float
-        The real harmonic $Y^m_n$ sampled at `theta` and `phi`
+        The real harmonic $Y^m_n$ sampled at `theta` and `phi` as
+        implemented in mrtrix. Warning: the basis is Tournier et al.
+        2007 [2]_; 2004 [1]_ is slightly different.
     m : array
-        The order of the harmonics.
-    n : array
         The degree of the harmonics.
+    n : array
+        The order of the harmonics.
 
     References
     ----------
-    .. [1] Descoteaux, M., Angelino, E., Fitzgibbons, S. and Deriche, R.
-           Regularized, Fast, and Robust Analytical Q-ball Imaging.
-           Magn. Reson. Med. 2007;58:497-510.
+    .. [1] Tournier J.D., Calamante F., Gadian D.G. and Connelly A.
+           Direct estimation of the fibre orientation density function from
+           diffusion-weighted MRI data using spherical deconvolution.
+           NeuroImage. 2004;23:1176-1185.
+    .. [2] Tournier J.D., Calamante F. and Connelly A. Robust determination
+           of the fibre orientation distribution in diffusion MRI:
+           Non-negativity constrained super-resolved spherical deconvolution.
+           NeuroImage. 2007;35(4):1459-1472.
 
     """
-    m, n = sph_harm_ind_list(sh_order)
-    phi = np.reshape(phi, [-1, 1])
-    theta = np.reshape(theta, [-1, 1])
-
-    real_sh = real_sph_harm(m, n, theta, phi)
-    return real_sh, m, n
+    deprecate_with_version('Function real_sym_sh_mrtix has been replaced \
```

Comment:
I recommend to : 
- use it as a decorator
- add from and until when

look [here](https://github.com/dipy/dipy/pull/2191/files#diff-3d2aafe93b8f41be9cf3521aa9d5d194R20-R24) as an example

Reply:
Hi @skoudoro, I fixed it in order to use it as a decorator in [11beb24](https://github.com/dipy/dipy/pull/2206/commits/11beb2414689604f8c256a4dfb1e703c34a70ff4)! As for the from and until, what values should I use?